src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/IamProxyAuthProvider.scala

```diff
@@ -120,24 +120,26 @@ class IamProxyAuthProvider(config: Config, serviceAccountProvider: ServiceAccoun
     * The returned future should complete once the provider has finished doing any associated work.
     * Leo will wait, so be timely!
     *
+    * @param internalId     The internal ID for the cluster (i.e. used for Sam resources)
     * @param creatorEmail     The email address of the user in question
     * @param googleProject The Google project the cluster was created in
     * @param clusterName   The user-provided name of the Dataproc cluster
     * @return A Future that will complete when the auth provider has finished doing its business.
     */
-  def notifyClusterCreated(creatorEmail: WorkbenchEmail, googleProject: GoogleProject, clusterName: ClusterName)(implicit executionContext: ExecutionContext): Future[Unit] = Future.successful(())
+  def notifyClusterCreated(internalId: String, creatorEmail: WorkbenchEmail, googleProject: GoogleProject, clusterName: ClusterName)(implicit executionContext: ExecutionContext): Future[Unit] = Future.successful(())
```

Comment:
if this is supposed to be UUID, I would use `java.util.UUID` for its type

Reply:
would love to, but that would require migrating existing clusters to have UUIDs as their sam resource IDs which is non-trivial

see https://github.com/DataBiosphere/leonardo/pull/1009/files#diff-b924c27cbfe0ca9ea81a34fca450e6f3R92

Reply:
but a new type, like `InternalId` might make sense here

Reply:
I ended up making a new type `ClusterInternalId`